tools/transform.cpp

```diff
@@ -301,8 +301,39 @@ TransformVerify::TransformVerify(Transform &t, bool check_each_var) :
       tgt_instrs.emplace(i.getName(), &i);
     }
   }
+}
+
+Errors TransformVerify::sync() {
+  try {
+    t.tgt.syncDataWithSrc(t.src);
+
+    // Check sizes of global variables
+    auto globals_tgt = t.tgt.getGlobalVars();
+    auto globals_src = t.src.getGlobalVars();
+    for (auto GVS: globals_src) {
+      auto I = find_if(globals_tgt.begin(), globals_tgt.end(),
+        [GVS](auto *GV) -> bool { return GVS->getName() == GV->getName(); });
+      if (I == globals_tgt.end())
+        continue;
 
-  t.tgt.syncDataWithSrc(t.src);
+      auto GVT = *I;
+      if (GVS->size() != GVT->size()) {
+        stringstream ss;
+        ss << "Global variable " << GVS->getName() << " in source and "
+          << "target has different sizes (" << GVS->size() << ", "
```

Comment:
maybe reshuffle the sentence a bit: Unsupported transformation: global variable foo has different size in source and target (xx vs yy bytes)

Reply:
Updated